src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java

```diff
@@ -0,0 +1,167 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check ensures that roundabouts with unreasonable valences are flagged. In reference to OSM
+ * Wiki, each roundabout should have greater than 1 connection as 1 connection should be tagged as a
+ * turning point, and no connections is obviously not a valid way.
+ *
+ * @author savannahostrowski
+ */
+public class RoundaboutValenceCheck extends BaseCheck
+{
+
+    private static final long serialVersionUID = 1L;
+    public static final String WRONG_VALENCE_INSTRUCTIONS = "This roundabout, {0,number,#}, "
+            + "has the wrong valence. It has a valence of {1,number,#}.";
+    public static final String VALENCE_OF_ONE_INSTRUCTIONS = "This feature, {0,number,#},"
+            + " should not be labelled as a roundabout. "
+            + "This feature should be a turning loop or turning circle.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(WRONG_VALENCE_INSTRUCTIONS, VALENCE_OF_ONE_INSTRUCTIONS);
+
+    private static final double LOWER_VALENCE_THRESHOLD_DEFAULT = 2.0;
+    private static final double UPPER_VALENCE_THRESHOLD_DEFAULT = 10.0;
+    private final double minimumValence;
+    private final double maximumValence;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public RoundaboutValenceCheck(final Configuration configuration)
+    {
+        super(configuration);
+
+        this.minimumValence = (double) configurationValue(configuration, "connections.minimum",
+                LOWER_VALENCE_THRESHOLD_DEFAULT);
+        this.maximumValence = (double) configurationValue(configuration, "connections.maximum",
+                UPPER_VALENCE_THRESHOLD_DEFAULT);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // We check that the object is an instance of Edge
+        return object instanceof Edge
+                // And that the Edge has not already been marked as flagged
+                && !this.isFlagged(object.getIdentifier())
+                // Make sure that the edges are instances of roundabout
+                && JunctionTag.isRoundabout(object);
```

Comment:
If an edge is encountered, then whether it is roundabout or not should be checked first. Because, check doesn't work on non-roundabout edges. This condition needs to filter first. Otherwise, many edges will be checked whether they are flagged or not first before checking if they are roundabout or not.

Reply:
You can further filter if an edge is master or not here.

Reply:
So you mean that `JunctionTag.isRoundabout(object)` and `!this.isFlagged(object.getIdentifier())` should be switched?

Nice! I like the idea of adding master edge filtering here.

Reply:
Yes, filter roundabouts first.